File: modules/allowance-questboard-edge/supabase/functions/quest/infrastructure/entity/quest_entity.ts
export class QuestEntity {
    constructor(
        public id: number,
        public title: string,
        public iconCode: string,
        public createdAt: Date,
        public updatedAt: Date,
    ) {}

    static fromMap(map: Record<string, unknown>): QuestEntity {
        return new QuestEntity(
            map["id"] as number,
            map["quest_title"] as string,
            map["icon_code"] as string,
            new Date(map["created_at"] as string),
            new Date(map["updated_at"] as string),
        );
    }

    toMap(): Record<string, unknown> {
        return {
            id: this.id,
            quest_title: this.title,
            icon_code: this.iconCode,
            created_at: this.createdAt.toISOString(),
            updated_at: this.updatedAt.toISOString(),
        };
    }
}

File: apps/allowance-questboard-flutter/lib/infrastracture/entity/quest_entity.dart
// import 'package:cloud_firestore/cloud_firestore.dart';

// class QuestEntity {
//   final String? id;
//   final String title;
//   final String client;
//   final String requestDetails;
//   final DateTime createdAt;
//   final DateTime updatedAt;

//   QuestEntity({
//     this.id,
//     required this.title,
//     required this.client,
//     required this.requestDetails,
//     required this.createdAt,
//     required this.updatedAt,
//   });

//   factory QuestEntity.fromMap(Map<String, dynamic> map, String documentId) {
//     return QuestEntity(
//         id: documentId,
//         title: map['title'],
//         client: map['client'] ?? '',
//         requestDetails: map['request_details'] ?? '',
//         createdAt: (map['created_at'] as Timestamp).toDate(),
//         updatedAt: (map['updated_at'] as Timestamp).toDate());
//   }

//   Map<String, dynamic> toMap() {
//     return {
//       'title': title,
//       'client': client,
//       'request_details': requestDetails,
//       'created_at': Timestamp.fromDate(createdAt),
//       'updated_at': Timestamp.fromDate(updatedAt),
//     };
//   }
// }
